sdk/state/open.go

```diff
@@ -110,6 +110,26 @@ func (c *Channel) ProposeOpen(p OpenParams) (OpenAgreement, error) {
 	return open, nil
 }
 
+func (c *Channel) validateOpen(m OpenAgreement) error {
+	// If the open agreement details don't match the open agreement in progress, error.
+	if !c.openAgreement.isEmpty() && !m.Details.Equal(c.openAgreement.Details) {
+		return fmt.Errorf("input open agreement details do not match the saved open agreement details")
+	}
+
+	// If the expiry of the agreement is past the max expiry the channel will accept, error.
+	if m.Details.ExpiresAt.After(time.Now().Add(c.maxOpenExpiry)) {
+		return fmt.Errorf("input open agreement expire too far into the future")
+	}
+
+	// If the open agreement has extra signatures, error.
+	if len(m.DeclarationSignatures) > 2 || len(m.CloseSignatures) > 2 || len(m.FormationSignatures) > 2 {
+		return fmt.Errorf("input open agreement has too many signatures, has declaration: %d,"+
+			" close: %d, formation: %d, max of 2 allowed for each",
+			len(m.DeclarationSignatures), len(m.CloseSignatures), len(m.FormationSignatures))
+	}
+	return nil
```

Comment:
I think I need to see the larger change it fits into to understand why we need the signature length check.

This validation is incomplete because this will only prevent a confirmation from having more than two at the start of the function, but if it has two remote signatures, then the local adds signatures, then it will have three. Maybe shift adding the signature count validation into the PR you're planning that'll verify for extra signatures?

I suspect signature validation will need to happen at the end, or we'll need to restructure verifying and signing signatures so it happens in a single pass, more like how we do it here: https://github.com/stellar/go/blob/90387a8/txnbuild/transaction.go#L1262-L1301.

Reply:
the larger change of checking extra signatures can be seen in this PR: https://github.com/stellar/experimental-payment-channels/pull/169 . Removing this dup code from this PR sorry about that.

But that's a great point about needing to check at the end I didn't think about that. Will make that change in that PR

Reply:
https://github.com/stellar/experimental-payment-channels/pull/183/commits/423700c2602323a4309bd19d3b65e6815fa1eb7d